src/core.c/CompUnit/Repository/Installation.rakumod

```diff
@@ -337,30 +337,26 @@ sub MAIN(:$name, :$auth, :$ver, *@, *%) {
             my $repo-prefix = self!repo-prefix;
             my $*DISTRIBUTION = CompUnit::Repository::Distribution.new($dist, :repo(self), :$dist-id);
             my $*RESOURCES = Distribution::Resources.new(:repo(self), :$dist-id);
-            my %done;
 
             my $compiler-id = CompUnit::PrecompilationId.new-without-check($*RAKU.compiler.id);
             for %provides.sort {
                 my $id = CompUnit::PrecompilationId.new-without-check($_.value.values[0]<file>);
                 $precomp.store.delete($compiler-id, $id);
             }
 
-            for %provides.sort {
-                my $id = $_.value.values[0]<file>;
+            my $to-precompile := %provides.map({ $_.key => $_.value.values[0]<file> }).sort(*.key).unique(:as(*.value));
```

Comment:
By using `unique` here we should avoid the need for `%done`. And since it is called before we use any parallelization we should always get the same outcome (such as when two different short-names have the same sha1).